<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Predicting cell states and their variability in single-cell or spatial omics data</title>
  <meta name="description" content="Predicting cell states and their variability in single-cell or spatial omics data" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Predicting cell states and their variability in single-cell or spatial omics data" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Predicting cell states and their variability in single-cell or spatial omics data" />
  
  
  



<meta name="date" content="2024-12-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="overview-of-svp.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="overview-of-svp.html"><a href="overview-of-svp.html"><i class="fa fa-check"></i><b>1</b> Overview of SVP</a></li>
<li class="chapter" data-level="2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html"><i class="fa fa-check"></i><b>2</b> Quantification of cell states using SVP</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cancersea-etc-function"><i class="fa fa-check"></i><b>2.1</b> Quantification of CancerSEA etc function</a></li>
<li class="chapter" data-level="2.2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type"><i class="fa fa-check"></i><b>2.2</b> Quantification of cell-type</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-pre-established-marker-gene-sets"><i class="fa fa-check"></i><b>2.2.1</b> Quantification of cell-type using the pre-established marker gene sets</a></li>
<li class="chapter" data-level="2.2.2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-reference-single-cell-data"><i class="fa fa-check"></i><b>2.2.2</b> Quantification of cell-type using the reference single-cell data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html"><i class="fa fa-check"></i><b>3</b> Spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#univariate-spatial-statistical-analysis"><i class="fa fa-check"></i><b>3.1</b> Univariate spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#identification-of-spatial-variable-features"><i class="fa fa-check"></i><b>3.1.1</b> Identification of spatial variable features</a></li>
<li class="chapter" data-level="3.1.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#identification-of-local-spatial-aggregation-areas"><i class="fa fa-check"></i><b>3.1.2</b> Identification of local spatial aggregation areas</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#bivariate-spatial-statistical-analysis"><i class="fa fa-check"></i><b>3.2</b> Bivariate spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#global-bivariate-spatial-analysis"><i class="fa fa-check"></i><b>3.2.1</b> Global bivariate spatial analysis</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#local-bivariate-spatial-analysis"><i class="fa fa-check"></i><b>3.2.2</b> Local bivariate spatial analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>4</b> Session information</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Predicting cell states and their variability in single-cell or spatial omics data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Predicting cell states and their variability in single-cell or spatial omics data</h1>
<p class="author multi-author"><em>Shuangbin Xu and GuangChuang Yu</em></p>
<address class="author_afil">
Department of Bioinformatics, School of Basic Medical Sciences, Southern Medical University<br>
<a class="author_email" href="mailto:#"><a href="mailto:xshuangbin@163.com" class="email">xshuangbin@163.com</a> and <a href="mailto:guangchuangyu@gmail.com" class="email">guangchuangyu@gmail.com</a></a>
</address>
<p class="date"><em>2024-12-13</em></p>
</div>
<div id="introduction" class="section level1 unnumbered hasAnchor">
<h1>Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Understanding the spatial distribution and interplay of cell states in tissue is critical for elucidating tissue formation and function. Single-cell and spatial omics present a promising approach to addressing this need. Traditional methods typically include the identification of highly variable genes, dimensionality reduction, clustering, and the annotation of cells or functions based on gene over-expression. Nevertheless, these qualitative approaches are inadequate for accurately mapping the distributions of spatial features. To address this, integrating biomedical knowledge such as Gene Ontology, KEGG, Reactome, transcription factors, and cell-type marker genes directly allows for the evaluation of cell states from gene expression data, creating quantitative functional pathway profiles at the single captured location.</p>
<p>After quantifying cell functions, analyzing their spatial distribution and co-distribution with other features can provide deeper insights into related biological questions. We focus on three aspects: the spatial variability of cell functions, regions where these functions cluster, and their co-distribution patterns with other features. Although existing tools such as <code>SPARK-X</code><span class="citation">(<a href="#ref-SPARKX">Zhu, Sun, and Zhou 2021</a>)</span>, <code>nnSVG</code><span class="citation">(<a href="#ref-nnSVG">Weber et al. 2023</a>)</span>, <code>SpatialDE</code><span class="citation">(<a href="#ref-SpatialDE">Svensson, Teichmann, and Stegle 2018</a>)</span>, <code>SpaGFT</code><span class="citation">(<a href="#ref-SpaGFT">Chang et al. 2024</a>)</span>, <code>Seurat</code><span class="citation">(<a href="#ref-Seurat">Hao et al. 2023</a>)</span>, and <code>Squidpy</code><span class="citation">(<a href="#ref-Squidpy">Palla et al. 2022</a>)</span> facilitate the exploration of spatially variable genes, they are primarily designed for gene-level analysis and lack the capability to investigate the spatial co-distribution of features. Additionally, many of these tools, including <code>SpatialDE</code><span class="citation">(<a href="#ref-SpatialDE">Svensson, Teichmann, and Stegle 2018</a>)</span>, <code>SPARK</code><span class="citation">(<a href="#ref-SPARK">Sun, Zhu, and Zhou 2020</a>)</span>, <code>MERINGUE</code><span class="citation">(<a href="#ref-MERINGUE">Miller et al. 2021</a>)</span>, and <code>nnSVG</code><span class="citation">(<a href="#ref-nnSVG">Weber et al. 2023</a>)</span>, face challenges in handling large-scale spatial transcriptome data due to high memory consumption and low computational efficiency.</p>
<p>To fill the gaps, we developed <code>SVP</code> to accurately predict cell states, explore their spatial distribution, and assess their spatial relationship with other features.</p>

</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-SpaGFT" class="csl-entry">
Chang, Yuzhou, Jixin Liu, Yi Jiang, Anjun Ma, Yao Yu Yeo, Qi Guo, Megan McNutt, et al. 2024. <span>“Graph Fourier Transform for Spatial Omics Representation and Analyses of Complex Organs.”</span> <em>Nature Communications</em> 15 (1): 7467.
</div>
<div id="ref-Seurat" class="csl-entry">
Hao, Yuhan, Tim Stuart, Madeline H Kowalski, Saket Choudhary, Paul Hoffman, Austin Hartman, Avi Srivastava, et al. 2023. <span>“Dictionary Learning for Integrative, Multimodal and Scalable Single-Cell Analysis.”</span> <em>Nature Biotechnology</em>. <a href="https://doi.org/10.1038/s41587-023-01767-y">https://doi.org/10.1038/s41587-023-01767-y</a>.
</div>
<div id="ref-MERINGUE" class="csl-entry">
Miller, Brendan F, Dhananjay Bambah-Mukku, Catherine Dulac, Xiaowei Zhuang, and Jean Fan. 2021. <span>“Characterizing Spatial Gene Expression Heterogeneity in Spatially Resolved Single-Cell Transcriptomic Data with Nonuniform Cellular Densities.”</span> <em>Genome Research</em> 31 (10): 1843–55.
</div>
<div id="ref-Squidpy" class="csl-entry">
Palla, Giovanni, Hannah Spitzer, Michal Klein, David Fischer, Anna Christina Schaar, Louis Benedikt Kuemmerle, Sergei Rybakov, et al. 2022. <span>“Squidpy: A Scalable Framework for Spatial Omics Analysis.”</span> <em>Nature Methods</em> 19 (2): 171–78. <a href="https://doi.org/10.1038/s41592-021-01358-2">https://doi.org/10.1038/s41592-021-01358-2</a>.
</div>
<div id="ref-SPARK" class="csl-entry">
Sun, Shiquan, Jiaqiang Zhu, and Xiang Zhou. 2020. <span>“Statistical Analysis of Spatial Expression Patterns for Spatially Resolved Transcriptomic Studies.”</span> <em>Nature Methods</em> 17 (2): 193–200.
</div>
<div id="ref-SpatialDE" class="csl-entry">
Svensson, Valentine, Sarah A Teichmann, and Oliver Stegle. 2018. <span>“SpatialDE: Identification of Spatially Variable Genes.”</span> <em>Nature Methods</em> 15 (5): 343–46.
</div>
<div id="ref-nnSVG" class="csl-entry">
Weber, Lukas M., Arkajyoti Saha, Abhirup Datta, Kasper D. Hansen, and Stephanie C. Hicks. 2023. <span>“nnSVG for the Scalable Identification of Spatially Variable Genes Using Nearest-Neighbor Gaussian Processes.”</span> <em>Nature Communications</em> 14 (4059). https://doi.org/<a href="https://doi.org/10.1038/s41467-023-39748-z">https://doi.org/10.1038/s41467-023-39748-z</a>.
</div>
<div id="ref-SPARKX" class="csl-entry">
Zhu, Jiaqiang, Shiquan Sun, and Xiang Zhou. 2021. <span>“SPARK-x: Non-Parametric Modeling Enables Scalable and Robust Detection of Spatial Expression Patterns for Large Spatial Transcriptomic Studies.”</span> <em>Genome Biology</em> 22 (1): 184. <a href="https://doi.org/10.1186/s13059-021-02404-0">https://doi.org/10.1186/s13059-021-02404-0</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="overview-of-svp.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
