---
title: "Predicting cell states and their variability in single-cell or spatial omics data"
author: 
- name: Shuangbin Xu and GuangChuang Yu
  email: xshuangbin@163.com and guangchuangyu@gmail.com
  affiliation: Department of Bioinformatics, School of Basic Medical Sciences, Southern Medical University
date: "`r Sys.Date()`"
knit: "bookdown::render_book"
site: bookdown::bookdown_site
output:
  bookdown::bs4_book:
  bookdown::pdf_book:
    dev: "cairo_pdf"
    latex_engine: xelatex
bibliography: references.bib
biblio-style: apalike
always_allow_html: yes
toc_appendix: yes
fontsize: "12pt"
mainfont: Times New Roman
sansfont: Arial
monofontoptions: "Scale=0.7"
linestretch: 1.5
toc-depth: 2
link-citations: yes
documentclass: book
papersize: A4
classoption: twoside
highlight_bw: yes
geometry: "left=35mm,right=35mm,top=25mm,bottom=25mm"
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r, echo = FALSE, results="hide", message=FALSE, loadpkgs}
{library(SingleCellExperiment)
library(SpatialExperiment)
library(ggsc)
library(scuttle)
library(scran)
library(SVP)
library(ggplot2)
library(magrittr)
library(scatterpie)
} |> suppressPackageStartupMessages()
```

# Introduction {-}

Understanding the spatial distribution and interplay of cell states in tissue is critical for elucidating tissue formation and function. Single-cell and spatial omics present a promising approach to addressing this need. Traditional methods typically include the identification of highly variable genes, dimensionality reduction, clustering, and the annotation of cells or functions based on gene over-expression. Nevertheless, these qualitative approaches are inadequate for accurately mapping the distributions of spatial features. To address this, integrating biomedical knowledge such as Gene Ontology, KEGG, Reactome, transcription factors, and cell-type marker genes directly allows for the evaluation of cell states from gene expression data, creating quantitative functional pathway profiles at the single captured location.

After quantifying cell functions, analyzing their spatial distribution and co-distribution with other features can provide deeper insights into related biological questions. We focus on three aspects: the spatial variability of cell functions, regions where these functions cluster, and their co-distribution patterns with other features. Although existing tools such as `SPARK-X`[@SPARKX], `nnSVG`[@nnSVG], `SpatialDE`[@SpatialDE], `SpaGFT`[@SpaGFT], `Seurat`[@Seurat], and `Squidpy`[@Squidpy] facilitate the exploration of spatially variable genes, they are primarily designed for gene-level analysis and lack the capability to investigate the spatial co-distribution of features. Additionally, many of these tools, including `SpatialDE`[@SpatialDE], `SPARK`[@SPARK], `MERINGUE`[@MERINGUE], and `nnSVG`[@nnSVG], face challenges in handling large-scale spatial transcriptome data due to high memory consumption and low computational efficiency.

To fill the gaps, we developed `SVP` to accurately predict cell states, explore their spatial distribution, and assess their spatial relationship with other features.

