<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1 Overview of SVP | Predicting cell states and their variability in single-cell or spatial omics data</title>
  <meta name="description" content="1 Overview of SVP | Predicting cell states and their variability in single-cell or spatial omics data" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="1 Overview of SVP | Predicting cell states and their variability in single-cell or spatial omics data" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1 Overview of SVP | Predicting cell states and their variability in single-cell or spatial omics data" />
  
  
  



<meta name="date" content="2024-12-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="quantification-of-cell-states-using-svp.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="overview-of-svp.html"><a href="overview-of-svp.html"><i class="fa fa-check"></i><b>1</b> Overview of SVP</a></li>
<li class="chapter" data-level="2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html"><i class="fa fa-check"></i><b>2</b> Quantification of cell states using SVP</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cancersea-etc-function"><i class="fa fa-check"></i><b>2.1</b> Quantification of CancerSEA etc function</a></li>
<li class="chapter" data-level="2.2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type"><i class="fa fa-check"></i><b>2.2</b> Quantification of cell-type</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-pre-established-marker-gene-sets"><i class="fa fa-check"></i><b>2.2.1</b> Quantification of cell-type using the pre-established marker gene sets</a></li>
<li class="chapter" data-level="2.2.2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-reference-single-cell-data"><i class="fa fa-check"></i><b>2.2.2</b> Quantification of cell-type using the reference single-cell data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html"><i class="fa fa-check"></i><b>3</b> Spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#univariate-spatial-statistical-analysis"><i class="fa fa-check"></i><b>3.1</b> Univariate spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#identification-of-spatial-variable-features"><i class="fa fa-check"></i><b>3.1.1</b> Identification of spatial variable features</a></li>
<li class="chapter" data-level="3.1.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#identification-of-local-spatial-aggregation-areas"><i class="fa fa-check"></i><b>3.1.2</b> Identification of local spatial aggregation areas</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#bivariate-spatial-statistical-analysis"><i class="fa fa-check"></i><b>3.2</b> Bivariate spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#global-bivariate-spatial-analysis"><i class="fa fa-check"></i><b>3.2.1</b> Global bivariate spatial analysis</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#local-bivariate-spatial-analysis"><i class="fa fa-check"></i><b>3.2.2</b> Local bivariate spatial analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>4</b> Session information</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Predicting cell states and their variability in single-cell or spatial omics data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="overview-of-svp" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">1</span> Overview of SVP<a href="overview-of-svp.html#overview-of-svp" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The evaluation of functional status at individual locations captured by SVP is achieved using Multiple Correspondence Analysis (<code>MCA</code>) for dimensionality reduction. This process employs a standardized gene expression matrix to project both cells and genes into a unified <code>MCA</code> space. It has been established that this method allows for the calculation of distances not only between genes and cells but also between cells and genes, thereby facilitating the assessment of their associations<span class="citation">(<a href="#ref-CelliD">Cortal et al. 2021</a>)</span>. Proximity in this space indicates a stronger relationship. These calculated distances are crucial for constructing a weighted k-nearest neighbors (KNN) network, linking each cell or gene to its most relevant counterparts. To discern features with varying levels of proximity, distances are first normalized to a 0-1 scale, with closer distances approaching 1 and farther distances approaching 0. This normalization is followed by division by the total distance among the nearest features, thereby assigning greater connection weights to closer features. Subsequently, databases of known biological knowledge, such as transcription factor target gene sets, Reactome functional gene sets, and cell-type marker gene sets, serve as initial seeds. Random walks on the constructed weighted KNN network yield preliminary functional state activity scores for each location. To mitigate potential biases introduced by dimensionality reduction, a hypergeometric distribution test enhances the enrichment analysis of top-ranking genes extracted directly from the expression matrix at each location. These analyses provide weights for functional activities, culminating in the derivation of functional activity scores at the single captured location (Fig. 1A).</p>
<p>To identify spatially variable cell functions, we first established a cell neighbor weight matrix based on spot locations using the Delaunay triangulation (default) or KNN algorithm. This weight matrix, alongside global autocorrelation analyses such as Moran’s <em>I</em> (default), Geary’s <em>C</em>, or Getis-Ord’s <em>G</em>, facilitated the identification of spatially variable cell functions or gene characteristics. Additionally, we utilized the same cell neighbor weight matrix and local spatial autocorrelation algorithms (Local Getis-Ord or Local Moran) to delineate the local spatial distribution of these variable features (Fig. 1B). To examine spatial co-distribution among cell functions, we designed a bivariate spatial global and local autocorrelation algorithm employing the Lee index. This approach enables the assessment of correlation between different cell characteristics in their spatial distribution (Fig. 1C).</p>

<div class="figure" style="text-align: center"><span style="display:block;" id="fig:overview"></span>
<img src="SVP_Framework.png" alt="Overview of SVP" width="760" />
<p class="caption">
Figure 1.1: <strong>Overview of SVP</strong>
</p>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-CelliD" class="csl-entry">
Cortal, Akira, Loredana Martignetti, Emmanuelle Six, and Antonio Rausell. 2021. <span>“Gene Signature Extraction and Cell Identity Recognition at the Single-Cell Level with Cell-ID.”</span> <em>Nature Biotechnology</em> 39 (9): 1095–1102.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quantification-of-cell-states-using-svp.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
