<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Quantification of cell states using SVP | Predicting cell states and their variability in single-cell or spatial omics data</title>
  <meta name="description" content="2 Quantification of cell states using SVP | Predicting cell states and their variability in single-cell or spatial omics data" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Quantification of cell states using SVP | Predicting cell states and their variability in single-cell or spatial omics data" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Quantification of cell states using SVP | Predicting cell states and their variability in single-cell or spatial omics data" />
  
  
  



<meta name="date" content="2024-12-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="overview-of-svp.html"/>
<link rel="next" href="spatial-statistical-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="overview-of-svp.html"><a href="overview-of-svp.html"><i class="fa fa-check"></i><b>1</b> Overview of SVP</a></li>
<li class="chapter" data-level="2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html"><i class="fa fa-check"></i><b>2</b> Quantification of cell states using SVP</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cancersea-etc-function"><i class="fa fa-check"></i><b>2.1</b> Quantification of CancerSEA etc function</a></li>
<li class="chapter" data-level="2.2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type"><i class="fa fa-check"></i><b>2.2</b> Quantification of cell-type</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-pre-established-marker-gene-sets"><i class="fa fa-check"></i><b>2.2.1</b> Quantification of cell-type using the pre-established marker gene sets</a></li>
<li class="chapter" data-level="2.2.2" data-path="quantification-of-cell-states-using-svp.html"><a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-reference-single-cell-data"><i class="fa fa-check"></i><b>2.2.2</b> Quantification of cell-type using the reference single-cell data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html"><i class="fa fa-check"></i><b>3</b> Spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#univariate-spatial-statistical-analysis"><i class="fa fa-check"></i><b>3.1</b> Univariate spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#identification-of-spatial-variable-features"><i class="fa fa-check"></i><b>3.1.1</b> Identification of spatial variable features</a></li>
<li class="chapter" data-level="3.1.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#identification-of-local-spatial-aggregation-areas"><i class="fa fa-check"></i><b>3.1.2</b> Identification of local spatial aggregation areas</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#bivariate-spatial-statistical-analysis"><i class="fa fa-check"></i><b>3.2</b> Bivariate spatial statistical analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#global-bivariate-spatial-analysis"><i class="fa fa-check"></i><b>3.2.1</b> Global bivariate spatial analysis</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-statistical-analysis.html"><a href="spatial-statistical-analysis.html#local-bivariate-spatial-analysis"><i class="fa fa-check"></i><b>3.2.2</b> Local bivariate spatial analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>4</b> Session information</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Predicting cell states and their variability in single-cell or spatial omics data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quantification-of-cell-states-using-svp" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Quantification of cell states using SVP<a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-states-using-svp" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>SVP can integrate gene or protein lists from existing biomedical databases like cell-type markers (PanglaoDB, CellMarkers), gene ontology (GO), molecular features (MSigDB), and pathways (KEGG, Reactome, Wikipathways), transcription factors, and disease ontology, to assess cell states. Of course, the cell-type markers also can be extracted from reference single cell data. In this vignette, we use the spatial transcriptome and single cell transcriptome of a human pancreatic ductal adenocarcinomas from the article<span class="citation">(<a href="#ref-PDAC">Moncada et al. 2020</a>)</span> to demonstrate. Here, we first perform the normalization by <code>logNormCounts</code> of <code>scuttle</code> and the <code>MCA</code> dimensionality reduction by <code>runMCA</code> of <code>SVP</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="quantification-of-cell-states-using-svp.html#cb1-1" tabindex="-1"></a><span class="co"># load the packages required in the vignette</span></span>
<span id="cb1-2"><a href="quantification-of-cell-states-using-svp.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(SpatialExperiment)</span>
<span id="cb1-3"><a href="quantification-of-cell-states-using-svp.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-4"><a href="quantification-of-cell-states-using-svp.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb1-5"><a href="quantification-of-cell-states-using-svp.html#cb1-5" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb1-6"><a href="quantification-of-cell-states-using-svp.html#cb1-6" tabindex="-1"></a><span class="fu">library</span>(SVP)</span>
<span id="cb1-7"><a href="quantification-of-cell-states-using-svp.html#cb1-7" tabindex="-1"></a><span class="fu">library</span>(ggsc)</span>
<span id="cb1-8"><a href="quantification-of-cell-states-using-svp.html#cb1-8" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-9"><a href="quantification-of-cell-states-using-svp.html#cb1-9" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-10"><a href="quantification-of-cell-states-using-svp.html#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="quantification-of-cell-states-using-svp.html#cb1-11" tabindex="-1"></a><span class="co"># load the spatial transcriptome, it is a SpatialExperiment object</span></span>
<span id="cb1-12"><a href="quantification-of-cell-states-using-svp.html#cb1-12" tabindex="-1"></a><span class="co"># to build the object, you can refer to the SpatialExperiment package</span></span>
<span id="cb1-13"><a href="quantification-of-cell-states-using-svp.html#cb1-13" tabindex="-1"></a>pdac_a_spe <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">url</span>(<span class="st">&quot;https://raw.githubusercontent.com/xiangpin/Data_for_Vignette_SVP/main/data/pdac_a_spe.rds&quot;</span>)) </span>
<span id="cb1-14"><a href="quantification-of-cell-states-using-svp.html#cb1-14" tabindex="-1"></a><span class="co"># we had removed the genes that did not express in any captured location.</span></span>
<span id="cb1-15"><a href="quantification-of-cell-states-using-svp.html#cb1-15" tabindex="-1"></a><span class="co"># First, we use logNormCounts of scuttle to normalize the spatial transcriptome</span></span>
<span id="cb1-16"><a href="quantification-of-cell-states-using-svp.html#cb1-16" tabindex="-1"></a>pdac_a_spe <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(pdac_a_spe)</span>
<span id="cb1-17"><a href="quantification-of-cell-states-using-svp.html#cb1-17" tabindex="-1"></a><span class="co"># Now the normalized counts was added to the assays of pdac_a_spe as `logcounts`</span></span>
<span id="cb1-18"><a href="quantification-of-cell-states-using-svp.html#cb1-18" tabindex="-1"></a><span class="co"># it can be extracted via logcounts(pdac_a_spe) or assay(pdac_a_spe, &#39;logcounts&#39;)</span></span>
<span id="cb1-19"><a href="quantification-of-cell-states-using-svp.html#cb1-19" tabindex="-1"></a></span>
<span id="cb1-20"><a href="quantification-of-cell-states-using-svp.html#cb1-20" tabindex="-1"></a><span class="co"># Next, we use `runMCA` of `SVP` to preform the MCA dimensionality reduction</span></span>
<span id="cb1-21"><a href="quantification-of-cell-states-using-svp.html#cb1-21" tabindex="-1"></a>pdac_a_spe <span class="ot">&lt;-</span> <span class="fu">runMCA</span>(pdac_a_spe, <span class="at">assay.type =</span> <span class="st">&#39;logcounts&#39;</span>)</span>
<span id="cb1-22"><a href="quantification-of-cell-states-using-svp.html#cb1-22" tabindex="-1"></a><span class="co">#&gt; Computing Fuzzy Matrix</span></span>
<span id="cb1-23"><a href="quantification-of-cell-states-using-svp.html#cb1-23" tabindex="-1"></a><span class="co">#&gt; Computing SVD</span></span>
<span id="cb1-24"><a href="quantification-of-cell-states-using-svp.html#cb1-24" tabindex="-1"></a><span class="co">#&gt; Computing Coordinates</span></span>
<span id="cb1-25"><a href="quantification-of-cell-states-using-svp.html#cb1-25" tabindex="-1"></a><span class="co"># The MCA result was added to the reducedDims, it can be extracted by reducedDim(pdac_a_spe, &#39;MCA&#39;)</span></span>
<span id="cb1-26"><a href="quantification-of-cell-states-using-svp.html#cb1-26" tabindex="-1"></a><span class="co"># more information can refer to SingleCellExperiment package</span></span>
<span id="cb1-27"><a href="quantification-of-cell-states-using-svp.html#cb1-27" tabindex="-1"></a>pdac_a_spe</span>
<span id="cb1-28"><a href="quantification-of-cell-states-using-svp.html#cb1-28" tabindex="-1"></a><span class="co">#&gt; class: SpatialExperiment </span></span>
<span id="cb1-29"><a href="quantification-of-cell-states-using-svp.html#cb1-29" tabindex="-1"></a><span class="co">#&gt; dim: 13160 428 </span></span>
<span id="cb1-30"><a href="quantification-of-cell-states-using-svp.html#cb1-30" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb1-31"><a href="quantification-of-cell-states-using-svp.html#cb1-31" tabindex="-1"></a><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span id="cb1-32"><a href="quantification-of-cell-states-using-svp.html#cb1-32" tabindex="-1"></a><span class="co">#&gt; rownames(13160): 1-Mar 1-Sep ... ZZEF1 ZZZ3</span></span>
<span id="cb1-33"><a href="quantification-of-cell-states-using-svp.html#cb1-33" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb1-34"><a href="quantification-of-cell-states-using-svp.html#cb1-34" tabindex="-1"></a><span class="co">#&gt; colnames(428): Spot1 Spot2 ... Spot427 Spot428</span></span>
<span id="cb1-35"><a href="quantification-of-cell-states-using-svp.html#cb1-35" tabindex="-1"></a><span class="co">#&gt; colData names(3): sample_id cluster_domain sizeFactor</span></span>
<span id="cb1-36"><a href="quantification-of-cell-states-using-svp.html#cb1-36" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(1): MCA</span></span>
<span id="cb1-37"><a href="quantification-of-cell-states-using-svp.html#cb1-37" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb1-38"><a href="quantification-of-cell-states-using-svp.html#cb1-38" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span>
<span id="cb1-39"><a href="quantification-of-cell-states-using-svp.html#cb1-39" tabindex="-1"></a><span class="co">#&gt; spatialCoords names(2) : x y</span></span>
<span id="cb1-40"><a href="quantification-of-cell-states-using-svp.html#cb1-40" tabindex="-1"></a><span class="co">#&gt; imgData names(1): sample_id</span></span></code></pre></div>
<div id="quantification-of-cancersea-etc-function" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Quantification of CancerSEA etc function<a href="quantification-of-cell-states-using-svp.html#quantification-of-cancersea-etc-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>SVP accepts <code>SingleCellExperiment</code> or <code>SpatialExperiment</code> object as input. The gene set should be a named list object, and the elements of the list must be included in the row names of the <code>SingleCellExperiment</code> or <code>SpatialExperiment</code> object. To quantifying the CancerSEA or other function for each captured location, the command <code>runSGSA</code> is used.</p>

<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="quantification-of-cell-states-using-svp.html#cb2-1" tabindex="-1"></a><span class="co"># This is gene list of Cancer Single-cell State Atlas to comprehensively decode distinct </span></span>
<span id="cb2-2"><a href="quantification-of-cell-states-using-svp.html#cb2-2" tabindex="-1"></a><span class="co"># functional states of cancer cells at single cell resolution.</span></span>
<span id="cb2-3"><a href="quantification-of-cell-states-using-svp.html#cb2-3" tabindex="-1"></a><span class="fu">data</span>(CancerSEASymbol)</span>
<span id="cb2-4"><a href="quantification-of-cell-states-using-svp.html#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="quantification-of-cell-states-using-svp.html#cb2-5" tabindex="-1"></a><span class="co"># It contains 14 types of cancer single-cell state, you can obtain </span></span>
<span id="cb2-6"><a href="quantification-of-cell-states-using-svp.html#cb2-6" tabindex="-1"></a><span class="co"># more information via ?CancerSEASymbol</span></span>
<span id="cb2-7"><a href="quantification-of-cell-states-using-svp.html#cb2-7" tabindex="-1"></a><span class="fu">names</span>(CancerSEASymbol)</span>
<span id="cb2-8"><a href="quantification-of-cell-states-using-svp.html#cb2-8" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;Angiogenesis&quot;    &quot;Apoptosis&quot;       &quot;Cell_Cycle&quot;      &quot;Differentiation&quot;</span></span>
<span id="cb2-9"><a href="quantification-of-cell-states-using-svp.html#cb2-9" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;DNA_damage&quot;      &quot;DNA_repair&quot;      &quot;EMT&quot;             &quot;Hypoxia&quot;        </span></span>
<span id="cb2-10"><a href="quantification-of-cell-states-using-svp.html#cb2-10" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;Inflammation&quot;    &quot;Invasion&quot;        &quot;Metastasis&quot;      &quot;Proliferation&quot;  </span></span>
<span id="cb2-11"><a href="quantification-of-cell-states-using-svp.html#cb2-11" tabindex="-1"></a><span class="co">#&gt; [13] &quot;Quiescence&quot;      &quot;Stemness&quot;</span></span>
<span id="cb2-12"><a href="quantification-of-cell-states-using-svp.html#cb2-12" tabindex="-1"></a></span>
<span id="cb2-13"><a href="quantification-of-cell-states-using-svp.html#cb2-13" tabindex="-1"></a>pdac_a_spe <span class="ot">&lt;-</span> <span class="fu">runSGSA</span>(pdac_a_spe, </span>
<span id="cb2-14"><a href="quantification-of-cell-states-using-svp.html#cb2-14" tabindex="-1"></a>                      <span class="at">gset.idx.list =</span> CancerSEASymbol, <span class="co"># The target gene set list</span></span>
<span id="cb2-15"><a href="quantification-of-cell-states-using-svp.html#cb2-15" tabindex="-1"></a>                      <span class="at">assay.type =</span> <span class="st">&#39;logcounts&#39;</span>, <span class="co"># The name of assays of gene profiler</span></span>
<span id="cb2-16"><a href="quantification-of-cell-states-using-svp.html#cb2-16" tabindex="-1"></a>                      <span class="at">gsvaExp.name =</span> <span class="st">&#39;CancerSEA&#39;</span> <span class="co"># The name of result to save to gsvaExps slot</span></span>
<span id="cb2-17"><a href="quantification-of-cell-states-using-svp.html#cb2-17" tabindex="-1"></a>              )</span>
<span id="cb2-18"><a href="quantification-of-cell-states-using-svp.html#cb2-18" tabindex="-1"></a><span class="co">#&gt; Building the nearest neighbor graph with the distance between features and</span></span>
<span id="cb2-19"><a href="quantification-of-cell-states-using-svp.html#cb2-19" tabindex="-1"></a><span class="co">#&gt; cells ...</span></span>
<span id="cb2-20"><a href="quantification-of-cell-states-using-svp.html#cb2-20" tabindex="-1"></a><span class="co">#&gt; elapsed time is 6.205000 seconds</span></span>
<span id="cb2-21"><a href="quantification-of-cell-states-using-svp.html#cb2-21" tabindex="-1"></a><span class="co">#&gt; Building the seed matrix using the gene set and the nearest neighbor graph for</span></span>
<span id="cb2-22"><a href="quantification-of-cell-states-using-svp.html#cb2-22" tabindex="-1"></a><span class="co">#&gt; random walk with restart ...</span></span>
<span id="cb2-23"><a href="quantification-of-cell-states-using-svp.html#cb2-23" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.043000 seconds</span></span>
<span id="cb2-24"><a href="quantification-of-cell-states-using-svp.html#cb2-24" tabindex="-1"></a><span class="co">#&gt; Calculating the affinity score using random walk with restart ...</span></span>
<span id="cb2-25"><a href="quantification-of-cell-states-using-svp.html#cb2-25" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.226000 seconds</span></span>
<span id="cb2-26"><a href="quantification-of-cell-states-using-svp.html#cb2-26" tabindex="-1"></a><span class="co">#&gt; Tidying the result of random walk with restart ...</span></span>
<span id="cb2-27"><a href="quantification-of-cell-states-using-svp.html#cb2-27" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.009000 seconds</span></span>
<span id="cb2-28"><a href="quantification-of-cell-states-using-svp.html#cb2-28" tabindex="-1"></a><span class="co">#&gt; ORA analysis ...</span></span>
<span id="cb2-29"><a href="quantification-of-cell-states-using-svp.html#cb2-29" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.011000 seconds</span></span>
<span id="cb2-30"><a href="quantification-of-cell-states-using-svp.html#cb2-30" tabindex="-1"></a></span>
<span id="cb2-31"><a href="quantification-of-cell-states-using-svp.html#cb2-31" tabindex="-1"></a><span class="co"># Then the result was added to gsvaExps to return a SVPExperiment object, the result</span></span>
<span id="cb2-32"><a href="quantification-of-cell-states-using-svp.html#cb2-32" tabindex="-1"></a><span class="co"># can be extracted with gsvaExp, you can view more information via help(gsvaExp).</span></span>
<span id="cb2-33"><a href="quantification-of-cell-states-using-svp.html#cb2-33" tabindex="-1"></a>pdac_a_spe</span>
<span id="cb2-34"><a href="quantification-of-cell-states-using-svp.html#cb2-34" tabindex="-1"></a><span class="co">#&gt; class: SVPExperiment </span></span>
<span id="cb2-35"><a href="quantification-of-cell-states-using-svp.html#cb2-35" tabindex="-1"></a><span class="co">#&gt; dim: 13160 428 </span></span>
<span id="cb2-36"><a href="quantification-of-cell-states-using-svp.html#cb2-36" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb2-37"><a href="quantification-of-cell-states-using-svp.html#cb2-37" tabindex="-1"></a><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span id="cb2-38"><a href="quantification-of-cell-states-using-svp.html#cb2-38" tabindex="-1"></a><span class="co">#&gt; rownames(13160): 1-Mar 1-Sep ... ZZEF1 ZZZ3</span></span>
<span id="cb2-39"><a href="quantification-of-cell-states-using-svp.html#cb2-39" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb2-40"><a href="quantification-of-cell-states-using-svp.html#cb2-40" tabindex="-1"></a><span class="co">#&gt; colnames(428): Spot1 Spot2 ... Spot427 Spot428</span></span>
<span id="cb2-41"><a href="quantification-of-cell-states-using-svp.html#cb2-41" tabindex="-1"></a><span class="co">#&gt; colData names(3): sample_id cluster_domain sizeFactor</span></span>
<span id="cb2-42"><a href="quantification-of-cell-states-using-svp.html#cb2-42" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(1): MCA</span></span>
<span id="cb2-43"><a href="quantification-of-cell-states-using-svp.html#cb2-43" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb2-44"><a href="quantification-of-cell-states-using-svp.html#cb2-44" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span>
<span id="cb2-45"><a href="quantification-of-cell-states-using-svp.html#cb2-45" tabindex="-1"></a><span class="co">#&gt; spatialCoords names(2) : x y</span></span>
<span id="cb2-46"><a href="quantification-of-cell-states-using-svp.html#cb2-46" tabindex="-1"></a><span class="co">#&gt; imgData names(0):</span></span>
<span id="cb2-47"><a href="quantification-of-cell-states-using-svp.html#cb2-47" tabindex="-1"></a><span class="co">#&gt; gsvaExps names(1) : CancerSEA</span></span>
<span id="cb2-48"><a href="quantification-of-cell-states-using-svp.html#cb2-48" tabindex="-1"></a></span>
<span id="cb2-49"><a href="quantification-of-cell-states-using-svp.html#cb2-49" tabindex="-1"></a><span class="fu">gsvaExpNames</span>(pdac_a_spe)</span>
<span id="cb2-50"><a href="quantification-of-cell-states-using-svp.html#cb2-50" tabindex="-1"></a><span class="co">#&gt; [1] &quot;CancerSEA&quot;</span></span>
<span id="cb2-51"><a href="quantification-of-cell-states-using-svp.html#cb2-51" tabindex="-1"></a></span>
<span id="cb2-52"><a href="quantification-of-cell-states-using-svp.html#cb2-52" tabindex="-1"></a><span class="co"># The result is also a SingleCellExperiment or SpatialExperiment.</span></span>
<span id="cb2-53"><a href="quantification-of-cell-states-using-svp.html#cb2-53" tabindex="-1"></a><span class="fu">gsvaExp</span>(pdac_a_spe, <span class="st">&#39;CancerSEA&#39;</span>)</span>
<span id="cb2-54"><a href="quantification-of-cell-states-using-svp.html#cb2-54" tabindex="-1"></a><span class="co">#&gt; class: SpatialExperiment </span></span>
<span id="cb2-55"><a href="quantification-of-cell-states-using-svp.html#cb2-55" tabindex="-1"></a><span class="co">#&gt; dim: 14 428 </span></span>
<span id="cb2-56"><a href="quantification-of-cell-states-using-svp.html#cb2-56" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb2-57"><a href="quantification-of-cell-states-using-svp.html#cb2-57" tabindex="-1"></a><span class="co">#&gt; assays(1): affi.score</span></span>
<span id="cb2-58"><a href="quantification-of-cell-states-using-svp.html#cb2-58" tabindex="-1"></a><span class="co">#&gt; rownames(14): Angiogenesis Apoptosis ... Quiescence Stemness</span></span>
<span id="cb2-59"><a href="quantification-of-cell-states-using-svp.html#cb2-59" tabindex="-1"></a><span class="co">#&gt; rowData names(4): exp.gene.num gset.gene.num gene.occurrence.rate</span></span>
<span id="cb2-60"><a href="quantification-of-cell-states-using-svp.html#cb2-60" tabindex="-1"></a><span class="co">#&gt;   geneSets</span></span>
<span id="cb2-61"><a href="quantification-of-cell-states-using-svp.html#cb2-61" tabindex="-1"></a><span class="co">#&gt; colnames(428): Spot1 Spot2 ... Spot427 Spot428</span></span>
<span id="cb2-62"><a href="quantification-of-cell-states-using-svp.html#cb2-62" tabindex="-1"></a><span class="co">#&gt; colData names(3): cluster_domain sizeFactor sample_id</span></span>
<span id="cb2-63"><a href="quantification-of-cell-states-using-svp.html#cb2-63" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(0):</span></span>
<span id="cb2-64"><a href="quantification-of-cell-states-using-svp.html#cb2-64" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb2-65"><a href="quantification-of-cell-states-using-svp.html#cb2-65" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span>
<span id="cb2-66"><a href="quantification-of-cell-states-using-svp.html#cb2-66" tabindex="-1"></a><span class="co">#&gt; spatialCoords names(2) : x y</span></span>
<span id="cb2-67"><a href="quantification-of-cell-states-using-svp.html#cb2-67" tabindex="-1"></a><span class="co">#&gt; imgData names(0):</span></span>
<span id="cb2-68"><a href="quantification-of-cell-states-using-svp.html#cb2-68" tabindex="-1"></a></span>
<span id="cb2-69"><a href="quantification-of-cell-states-using-svp.html#cb2-69" tabindex="-1"></a><span class="co"># We use sc_spatial of ggsc to visualize the distribution of each functions</span></span>
<span id="cb2-70"><a href="quantification-of-cell-states-using-svp.html#cb2-70" tabindex="-1"></a><span class="co"># Note: when the number of features is too large, we reco</span></span>
<span id="cb2-71"><a href="quantification-of-cell-states-using-svp.html#cb2-71" tabindex="-1"></a><span class="fu">gsvaExp</span>(pdac_a_spe, <span class="st">&#39;CancerSEA&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-72"><a href="quantification-of-cell-states-using-svp.html#cb2-72" tabindex="-1"></a>  <span class="fu">sc_spatial</span>(</span>
<span id="cb2-73"><a href="quantification-of-cell-states-using-svp.html#cb2-73" tabindex="-1"></a>    <span class="at">features =</span> <span class="fu">rownames</span>(.),</span>
<span id="cb2-74"><a href="quantification-of-cell-states-using-svp.html#cb2-74" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y),</span>
<span id="cb2-75"><a href="quantification-of-cell-states-using-svp.html#cb2-75" tabindex="-1"></a>    <span class="at">geom =</span> geom_bgpoint,</span>
<span id="cb2-76"><a href="quantification-of-cell-states-using-svp.html#cb2-76" tabindex="-1"></a>    <span class="at">pointsize =</span> <span class="fl">1.2</span>,</span>
<span id="cb2-77"><a href="quantification-of-cell-states-using-svp.html#cb2-77" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">4</span>,</span>
<span id="cb2-78"><a href="quantification-of-cell-states-using-svp.html#cb2-78" tabindex="-1"></a>    <span class="at">common.legend =</span> <span class="cn">FALSE</span> <span class="co"># The output is a patchwork </span></span>
<span id="cb2-79"><a href="quantification-of-cell-states-using-svp.html#cb2-79" tabindex="-1"></a>  ) <span class="sc">&amp;</span> </span>
<span id="cb2-80"><a href="quantification-of-cell-states-using-svp.html#cb2-80" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option=</span><span class="st">&#39;B&#39;</span>, <span class="at">guide=</span><span class="st">&#39;none&#39;</span>)</span>
<span id="cb2-81"><a href="quantification-of-cell-states-using-svp.html#cb2-81" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-82"><a href="quantification-of-cell-states-using-svp.html#cb2-82" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-83"><a href="quantification-of-cell-states-using-svp.html#cb2-83" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-84"><a href="quantification-of-cell-states-using-svp.html#cb2-84" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-85"><a href="quantification-of-cell-states-using-svp.html#cb2-85" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-86"><a href="quantification-of-cell-states-using-svp.html#cb2-86" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-87"><a href="quantification-of-cell-states-using-svp.html#cb2-87" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-88"><a href="quantification-of-cell-states-using-svp.html#cb2-88" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-89"><a href="quantification-of-cell-states-using-svp.html#cb2-89" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-90"><a href="quantification-of-cell-states-using-svp.html#cb2-90" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-91"><a href="quantification-of-cell-states-using-svp.html#cb2-91" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-92"><a href="quantification-of-cell-states-using-svp.html#cb2-92" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-93"><a href="quantification-of-cell-states-using-svp.html#cb2-93" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-94"><a href="quantification-of-cell-states-using-svp.html#cb2-94" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-95"><a href="quantification-of-cell-states-using-svp.html#cb2-95" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-96"><a href="quantification-of-cell-states-using-svp.html#cb2-96" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-97"><a href="quantification-of-cell-states-using-svp.html#cb2-97" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-98"><a href="quantification-of-cell-states-using-svp.html#cb2-98" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-99"><a href="quantification-of-cell-states-using-svp.html#cb2-99" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-100"><a href="quantification-of-cell-states-using-svp.html#cb2-100" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-101"><a href="quantification-of-cell-states-using-svp.html#cb2-101" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-102"><a href="quantification-of-cell-states-using-svp.html#cb2-102" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-103"><a href="quantification-of-cell-states-using-svp.html#cb2-103" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-104"><a href="quantification-of-cell-states-using-svp.html#cb2-104" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-105"><a href="quantification-of-cell-states-using-svp.html#cb2-105" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-106"><a href="quantification-of-cell-states-using-svp.html#cb2-106" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span>
<span id="cb2-107"><a href="quantification-of-cell-states-using-svp.html#cb2-107" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb2-108"><a href="quantification-of-cell-states-using-svp.html#cb2-108" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:CancerSEA"></span>
<img src="SVP_files/figure-html/CancerSEA-1.png" alt="The heatmap of each CancerSEA function" width="1056" />
<p class="caption">
Figure 2.1: <strong>The heatmap of each CancerSEA function</strong>
</p>
</div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="quantification-of-cell-states-using-svp.html#cb3-1" tabindex="-1"></a></span>
<span id="cb3-2"><a href="quantification-of-cell-states-using-svp.html#cb3-2" tabindex="-1"></a><span class="co"># The other function also can be quantified with runSGSA</span></span>
<span id="cb3-3"><a href="quantification-of-cell-states-using-svp.html#cb3-3" tabindex="-1"></a><span class="co"># the gset.idx.list supports the named gene list, gson object or</span></span>
<span id="cb3-4"><a href="quantification-of-cell-states-using-svp.html#cb3-4" tabindex="-1"></a><span class="co"># locate gmt file or html url of gmt.</span></span>
<span id="cb3-5"><a href="quantification-of-cell-states-using-svp.html#cb3-5" tabindex="-1"></a>pdac_a_spe <span class="ot">&lt;-</span> <span class="fu">runSGSA</span>(</span>
<span id="cb3-6"><a href="quantification-of-cell-states-using-svp.html#cb3-6" tabindex="-1"></a>    pdac_a_spe, </span>
<span id="cb3-7"><a href="quantification-of-cell-states-using-svp.html#cb3-7" tabindex="-1"></a>    <span class="at">gset.idx.list =</span> <span class="st">&quot;https://data.broadinstitute.org/gsea-msigdb/msigdb/release/2023.2.Hs/h.all.v2023.2.Hs.symbols.gmt&quot;</span>,</span>
<span id="cb3-8"><a href="quantification-of-cell-states-using-svp.html#cb3-8" tabindex="-1"></a>    <span class="at">gsvaExp.name =</span> <span class="st">&#39;hallmark&#39;</span>,</span>
<span id="cb3-9"><a href="quantification-of-cell-states-using-svp.html#cb3-9" tabindex="-1"></a>    <span class="at">assay.type =</span> <span class="st">&#39;logcounts&#39;</span> <span class="co"># default</span></span>
<span id="cb3-10"><a href="quantification-of-cell-states-using-svp.html#cb3-10" tabindex="-1"></a>)</span>
<span id="cb3-11"><a href="quantification-of-cell-states-using-svp.html#cb3-11" tabindex="-1"></a><span class="co">#&gt; Building the nearest neighbor graph with the distance between features and</span></span>
<span id="cb3-12"><a href="quantification-of-cell-states-using-svp.html#cb3-12" tabindex="-1"></a><span class="co">#&gt; cells ...</span></span>
<span id="cb3-13"><a href="quantification-of-cell-states-using-svp.html#cb3-13" tabindex="-1"></a><span class="co">#&gt; elapsed time is 5.897000 seconds</span></span>
<span id="cb3-14"><a href="quantification-of-cell-states-using-svp.html#cb3-14" tabindex="-1"></a><span class="co">#&gt; Building the seed matrix using the gene set and the nearest neighbor graph for</span></span>
<span id="cb3-15"><a href="quantification-of-cell-states-using-svp.html#cb3-15" tabindex="-1"></a><span class="co">#&gt; random walk with restart ...</span></span>
<span id="cb3-16"><a href="quantification-of-cell-states-using-svp.html#cb3-16" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.030000 seconds</span></span>
<span id="cb3-17"><a href="quantification-of-cell-states-using-svp.html#cb3-17" tabindex="-1"></a><span class="co">#&gt; Calculating the affinity score using random walk with restart ...</span></span>
<span id="cb3-18"><a href="quantification-of-cell-states-using-svp.html#cb3-18" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.288000 seconds</span></span>
<span id="cb3-19"><a href="quantification-of-cell-states-using-svp.html#cb3-19" tabindex="-1"></a><span class="co">#&gt; Tidying the result of random walk with restart ...</span></span>
<span id="cb3-20"><a href="quantification-of-cell-states-using-svp.html#cb3-20" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.018000 seconds</span></span>
<span id="cb3-21"><a href="quantification-of-cell-states-using-svp.html#cb3-21" tabindex="-1"></a><span class="co">#&gt; ORA analysis ...</span></span>
<span id="cb3-22"><a href="quantification-of-cell-states-using-svp.html#cb3-22" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.028000 seconds</span></span>
<span id="cb3-23"><a href="quantification-of-cell-states-using-svp.html#cb3-23" tabindex="-1"></a></span>
<span id="cb3-24"><a href="quantification-of-cell-states-using-svp.html#cb3-24" tabindex="-1"></a><span class="co"># The result also was added to gsvaExps</span></span>
<span id="cb3-25"><a href="quantification-of-cell-states-using-svp.html#cb3-25" tabindex="-1"></a>pdac_a_spe </span>
<span id="cb3-26"><a href="quantification-of-cell-states-using-svp.html#cb3-26" tabindex="-1"></a><span class="co">#&gt; class: SVPExperiment </span></span>
<span id="cb3-27"><a href="quantification-of-cell-states-using-svp.html#cb3-27" tabindex="-1"></a><span class="co">#&gt; dim: 13160 428 </span></span>
<span id="cb3-28"><a href="quantification-of-cell-states-using-svp.html#cb3-28" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb3-29"><a href="quantification-of-cell-states-using-svp.html#cb3-29" tabindex="-1"></a><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span id="cb3-30"><a href="quantification-of-cell-states-using-svp.html#cb3-30" tabindex="-1"></a><span class="co">#&gt; rownames(13160): 1-Mar 1-Sep ... ZZEF1 ZZZ3</span></span>
<span id="cb3-31"><a href="quantification-of-cell-states-using-svp.html#cb3-31" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb3-32"><a href="quantification-of-cell-states-using-svp.html#cb3-32" tabindex="-1"></a><span class="co">#&gt; colnames(428): Spot1 Spot2 ... Spot427 Spot428</span></span>
<span id="cb3-33"><a href="quantification-of-cell-states-using-svp.html#cb3-33" tabindex="-1"></a><span class="co">#&gt; colData names(3): sample_id cluster_domain sizeFactor</span></span>
<span id="cb3-34"><a href="quantification-of-cell-states-using-svp.html#cb3-34" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(1): MCA</span></span>
<span id="cb3-35"><a href="quantification-of-cell-states-using-svp.html#cb3-35" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb3-36"><a href="quantification-of-cell-states-using-svp.html#cb3-36" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span>
<span id="cb3-37"><a href="quantification-of-cell-states-using-svp.html#cb3-37" tabindex="-1"></a><span class="co">#&gt; spatialCoords names(2) : x y</span></span>
<span id="cb3-38"><a href="quantification-of-cell-states-using-svp.html#cb3-38" tabindex="-1"></a><span class="co">#&gt; imgData names(0):</span></span>
<span id="cb3-39"><a href="quantification-of-cell-states-using-svp.html#cb3-39" tabindex="-1"></a><span class="co">#&gt; gsvaExps names(2) : CancerSEA hallmark</span></span></code></pre></div>
</div>
<div id="quantification-of-cell-type" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Quantification of cell-type<a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To quantifying the cell-type for each captured location, the gene set should be the marker gene sets of cell-type. It can be the pre-established marker gene sets, such as the markers from CellMarkers or PanglaoDB. High-quality cell marker genes from past reports or your own compilations are acceptable. In addition, It can also be extracted from the reference single cell data. The following sections will detail the specific operations of each approach.</p>
<div id="quantification-of-cell-type-using-the-pre-established-marker-gene-sets" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Quantification of cell-type using the pre-established marker gene sets<a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-pre-established-marker-gene-sets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can use the pre-established marker gene sets to quantifying the cell-type for each captured location. Here, we used the markers obtained from the <code>CARD</code>, which was used as input for <code>CARDfree</code> command.</p>
<p>(ref:runSGSA_free) <strong>The pie plot of cell-type activity</strong></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="quantification-of-cell-states-using-svp.html#cb4-1" tabindex="-1"></a><span class="co"># The marker gene sets</span></span>
<span id="cb4-2"><a href="quantification-of-cell-states-using-svp.html#cb4-2" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">url</span>(<span class="st">&quot;https://github.com/YingMa0107/CARD/blob/master/data/markerList.RData?raw=true&quot;</span>))</span>
<span id="cb4-3"><a href="quantification-of-cell-states-using-svp.html#cb4-3" tabindex="-1"></a>markerList <span class="sc">|&gt;</span> <span class="fu">names</span>()</span>
<span id="cb4-4"><a href="quantification-of-cell-states-using-svp.html#cb4-4" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;Acinar_cells&quot;                        </span></span>
<span id="cb4-5"><a href="quantification-of-cell-states-using-svp.html#cb4-5" tabindex="-1"></a><span class="co">#&gt;  [2] &quot;Ductal_terminal_ductal_like&quot;         </span></span>
<span id="cb4-6"><a href="quantification-of-cell-states-using-svp.html#cb4-6" tabindex="-1"></a><span class="co">#&gt;  [3] &quot;Ductal_CRISP3_high-centroacinar_like&quot;</span></span>
<span id="cb4-7"><a href="quantification-of-cell-states-using-svp.html#cb4-7" tabindex="-1"></a><span class="co">#&gt;  [4] &quot;Cancer_clone_A&quot;                      </span></span>
<span id="cb4-8"><a href="quantification-of-cell-states-using-svp.html#cb4-8" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;Ductal_MHC_Class_II&quot;                 </span></span>
<span id="cb4-9"><a href="quantification-of-cell-states-using-svp.html#cb4-9" tabindex="-1"></a><span class="co">#&gt;  [6] &quot;Cancer_clone_B&quot;                      </span></span>
<span id="cb4-10"><a href="quantification-of-cell-states-using-svp.html#cb4-10" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;mDCs_A&quot;                              </span></span>
<span id="cb4-11"><a href="quantification-of-cell-states-using-svp.html#cb4-11" tabindex="-1"></a><span class="co">#&gt;  [8] &quot;Ductal_APOL1_high-hypoxic&quot;           </span></span>
<span id="cb4-12"><a href="quantification-of-cell-states-using-svp.html#cb4-12" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;Tuft_cells&quot;                          </span></span>
<span id="cb4-13"><a href="quantification-of-cell-states-using-svp.html#cb4-13" tabindex="-1"></a><span class="co">#&gt; [10] &quot;mDCs_B&quot;                              </span></span>
<span id="cb4-14"><a href="quantification-of-cell-states-using-svp.html#cb4-14" tabindex="-1"></a><span class="co">#&gt; [11] &quot;pDCs&quot;                                </span></span>
<span id="cb4-15"><a href="quantification-of-cell-states-using-svp.html#cb4-15" tabindex="-1"></a><span class="co">#&gt; [12] &quot;Endocrine_cells&quot;                     </span></span>
<span id="cb4-16"><a href="quantification-of-cell-states-using-svp.html#cb4-16" tabindex="-1"></a><span class="co">#&gt; [13] &quot;Endothelial_cells&quot;                   </span></span>
<span id="cb4-17"><a href="quantification-of-cell-states-using-svp.html#cb4-17" tabindex="-1"></a><span class="co">#&gt; [14] &quot;Macrophages_A&quot;                       </span></span>
<span id="cb4-18"><a href="quantification-of-cell-states-using-svp.html#cb4-18" tabindex="-1"></a><span class="co">#&gt; [15] &quot;Mast_cells&quot;                          </span></span>
<span id="cb4-19"><a href="quantification-of-cell-states-using-svp.html#cb4-19" tabindex="-1"></a><span class="co">#&gt; [16] &quot;Macrophages_B&quot;                       </span></span>
<span id="cb4-20"><a href="quantification-of-cell-states-using-svp.html#cb4-20" tabindex="-1"></a><span class="co">#&gt; [17] &quot;T_cells_&amp;_NK_cells&quot;                  </span></span>
<span id="cb4-21"><a href="quantification-of-cell-states-using-svp.html#cb4-21" tabindex="-1"></a><span class="co">#&gt; [18] &quot;Monocytes&quot;                           </span></span>
<span id="cb4-22"><a href="quantification-of-cell-states-using-svp.html#cb4-22" tabindex="-1"></a><span class="co">#&gt; [19] &quot;RBCs&quot;                                </span></span>
<span id="cb4-23"><a href="quantification-of-cell-states-using-svp.html#cb4-23" tabindex="-1"></a><span class="co">#&gt; [20] &quot;Fibroblasts&quot;</span></span>
<span id="cb4-24"><a href="quantification-of-cell-states-using-svp.html#cb4-24" tabindex="-1"></a><span class="co"># We can view the marker gene number of each cell-type.</span></span>
<span id="cb4-25"><a href="quantification-of-cell-states-using-svp.html#cb4-25" tabindex="-1"></a><span class="fu">lapply</span>(markerList, length) <span class="sc">|&gt;</span> <span class="fu">unlist</span>()</span>
<span id="cb4-26"><a href="quantification-of-cell-states-using-svp.html#cb4-26" tabindex="-1"></a><span class="co">#&gt;                         Acinar_cells          Ductal_terminal_ductal_like </span></span>
<span id="cb4-27"><a href="quantification-of-cell-states-using-svp.html#cb4-27" tabindex="-1"></a><span class="co">#&gt;                                   49                                   13 </span></span>
<span id="cb4-28"><a href="quantification-of-cell-states-using-svp.html#cb4-28" tabindex="-1"></a><span class="co">#&gt; Ductal_CRISP3_high-centroacinar_like                       Cancer_clone_A </span></span>
<span id="cb4-29"><a href="quantification-of-cell-states-using-svp.html#cb4-29" tabindex="-1"></a><span class="co">#&gt;                                   34                                   81 </span></span>
<span id="cb4-30"><a href="quantification-of-cell-states-using-svp.html#cb4-30" tabindex="-1"></a><span class="co">#&gt;                  Ductal_MHC_Class_II                       Cancer_clone_B </span></span>
<span id="cb4-31"><a href="quantification-of-cell-states-using-svp.html#cb4-31" tabindex="-1"></a><span class="co">#&gt;                                    7                                  110 </span></span>
<span id="cb4-32"><a href="quantification-of-cell-states-using-svp.html#cb4-32" tabindex="-1"></a><span class="co">#&gt;                               mDCs_A            Ductal_APOL1_high-hypoxic </span></span>
<span id="cb4-33"><a href="quantification-of-cell-states-using-svp.html#cb4-33" tabindex="-1"></a><span class="co">#&gt;                                  292                                   47 </span></span>
<span id="cb4-34"><a href="quantification-of-cell-states-using-svp.html#cb4-34" tabindex="-1"></a><span class="co">#&gt;                           Tuft_cells                               mDCs_B </span></span>
<span id="cb4-35"><a href="quantification-of-cell-states-using-svp.html#cb4-35" tabindex="-1"></a><span class="co">#&gt;                                  128                                  209 </span></span>
<span id="cb4-36"><a href="quantification-of-cell-states-using-svp.html#cb4-36" tabindex="-1"></a><span class="co">#&gt;                                 pDCs                      Endocrine_cells </span></span>
<span id="cb4-37"><a href="quantification-of-cell-states-using-svp.html#cb4-37" tabindex="-1"></a><span class="co">#&gt;                                  134                                  182 </span></span>
<span id="cb4-38"><a href="quantification-of-cell-states-using-svp.html#cb4-38" tabindex="-1"></a><span class="co">#&gt;                    Endothelial_cells                        Macrophages_A </span></span>
<span id="cb4-39"><a href="quantification-of-cell-states-using-svp.html#cb4-39" tabindex="-1"></a><span class="co">#&gt;                                  165                                  210 </span></span>
<span id="cb4-40"><a href="quantification-of-cell-states-using-svp.html#cb4-40" tabindex="-1"></a><span class="co">#&gt;                           Mast_cells                        Macrophages_B </span></span>
<span id="cb4-41"><a href="quantification-of-cell-states-using-svp.html#cb4-41" tabindex="-1"></a><span class="co">#&gt;                                  286                                  167 </span></span>
<span id="cb4-42"><a href="quantification-of-cell-states-using-svp.html#cb4-42" tabindex="-1"></a><span class="co">#&gt;                   T_cells_&amp;_NK_cells                            Monocytes </span></span>
<span id="cb4-43"><a href="quantification-of-cell-states-using-svp.html#cb4-43" tabindex="-1"></a><span class="co">#&gt;                                   89                                  160 </span></span>
<span id="cb4-44"><a href="quantification-of-cell-states-using-svp.html#cb4-44" tabindex="-1"></a><span class="co">#&gt;                                 RBCs                          Fibroblasts </span></span>
<span id="cb4-45"><a href="quantification-of-cell-states-using-svp.html#cb4-45" tabindex="-1"></a><span class="co">#&gt;                                   92                                  224</span></span>
<span id="cb4-46"><a href="quantification-of-cell-states-using-svp.html#cb4-46" tabindex="-1"></a></span>
<span id="cb4-47"><a href="quantification-of-cell-states-using-svp.html#cb4-47" tabindex="-1"></a>pdac_a_spe <span class="ot">&lt;-</span> <span class="fu">runSGSA</span>(</span>
<span id="cb4-48"><a href="quantification-of-cell-states-using-svp.html#cb4-48" tabindex="-1"></a>                pdac_a_spe, </span>
<span id="cb4-49"><a href="quantification-of-cell-states-using-svp.html#cb4-49" tabindex="-1"></a>                <span class="at">gset.idx.list =</span> markerList,</span>
<span id="cb4-50"><a href="quantification-of-cell-states-using-svp.html#cb4-50" tabindex="-1"></a>                <span class="at">gsvaExp.name =</span> <span class="st">&#39;CellTypeFree&#39;</span>, </span>
<span id="cb4-51"><a href="quantification-of-cell-states-using-svp.html#cb4-51" tabindex="-1"></a>                <span class="at">assay.type =</span> <span class="st">&#39;logcounts&#39;</span></span>
<span id="cb4-52"><a href="quantification-of-cell-states-using-svp.html#cb4-52" tabindex="-1"></a>              )</span>
<span id="cb4-53"><a href="quantification-of-cell-states-using-svp.html#cb4-53" tabindex="-1"></a><span class="co">#&gt; Building the nearest neighbor graph with the distance between features and</span></span>
<span id="cb4-54"><a href="quantification-of-cell-states-using-svp.html#cb4-54" tabindex="-1"></a><span class="co">#&gt; cells ...</span></span>
<span id="cb4-55"><a href="quantification-of-cell-states-using-svp.html#cb4-55" tabindex="-1"></a><span class="co">#&gt; elapsed time is 6.086000 seconds</span></span>
<span id="cb4-56"><a href="quantification-of-cell-states-using-svp.html#cb4-56" tabindex="-1"></a><span class="co">#&gt; Building the seed matrix using the gene set and the nearest neighbor graph for</span></span>
<span id="cb4-57"><a href="quantification-of-cell-states-using-svp.html#cb4-57" tabindex="-1"></a><span class="co">#&gt; random walk with restart ...</span></span>
<span id="cb4-58"><a href="quantification-of-cell-states-using-svp.html#cb4-58" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.018000 seconds</span></span>
<span id="cb4-59"><a href="quantification-of-cell-states-using-svp.html#cb4-59" tabindex="-1"></a><span class="co">#&gt; Calculating the affinity score using random walk with restart ...</span></span>
<span id="cb4-60"><a href="quantification-of-cell-states-using-svp.html#cb4-60" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.234000 seconds</span></span>
<span id="cb4-61"><a href="quantification-of-cell-states-using-svp.html#cb4-61" tabindex="-1"></a><span class="co">#&gt; Tidying the result of random walk with restart ...</span></span>
<span id="cb4-62"><a href="quantification-of-cell-states-using-svp.html#cb4-62" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.012000 seconds</span></span>
<span id="cb4-63"><a href="quantification-of-cell-states-using-svp.html#cb4-63" tabindex="-1"></a><span class="co">#&gt; ORA analysis ...</span></span>
<span id="cb4-64"><a href="quantification-of-cell-states-using-svp.html#cb4-64" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.016000 seconds</span></span>
<span id="cb4-65"><a href="quantification-of-cell-states-using-svp.html#cb4-65" tabindex="-1"></a></span>
<span id="cb4-66"><a href="quantification-of-cell-states-using-svp.html#cb4-66" tabindex="-1"></a><span class="co"># Then we can visualize the cell-type activity via sc_spatial of SVP with pie plot</span></span>
<span id="cb4-67"><a href="quantification-of-cell-states-using-svp.html#cb4-67" tabindex="-1"></a><span class="fu">gsvaExp</span>(pdac_a_spe, <span class="st">&quot;CellTypeFree&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb4-68"><a href="quantification-of-cell-states-using-svp.html#cb4-68" tabindex="-1"></a>   <span class="fu">sc_spatial</span>(</span>
<span id="cb4-69"><a href="quantification-of-cell-states-using-svp.html#cb4-69" tabindex="-1"></a>     <span class="at">features =</span> <span class="fu">sort</span>(<span class="fu">names</span>(markerList)),</span>
<span id="cb4-70"><a href="quantification-of-cell-states-using-svp.html#cb4-70" tabindex="-1"></a>     <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y),</span>
<span id="cb4-71"><a href="quantification-of-cell-states-using-svp.html#cb4-71" tabindex="-1"></a>     <span class="at">plot.pie =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-72"><a href="quantification-of-cell-states-using-svp.html#cb4-72" tabindex="-1"></a>     <span class="at">color =</span> <span class="cn">NA</span>,</span>
<span id="cb4-73"><a href="quantification-of-cell-states-using-svp.html#cb4-73" tabindex="-1"></a>     <span class="at">pie.radius.scale =</span> .<span class="dv">9</span></span>
<span id="cb4-74"><a href="quantification-of-cell-states-using-svp.html#cb4-74" tabindex="-1"></a>   )</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="SVP_files/figure-html/runSGSA_free-1.png" alt="(ref:runSGSA_free)" width="864" />
<p class="caption">
(#fig:runSGSA_free)(ref:runSGSA_free)
</p>
</div>
<p>We cal also display each cell-type activity with heat-map according to the spatial coordinate.</p>
<p>(ref:heatmap_celltype) <strong>the heatmap of each cell-type activity</strong></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="quantification-of-cell-states-using-svp.html#cb5-1" tabindex="-1"></a>pdac_a_spe <span class="sc">|&gt;</span> <span class="fu">gsvaExp</span>(<span class="dv">3</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-2"><a href="quantification-of-cell-states-using-svp.html#cb5-2" tabindex="-1"></a>   <span class="fu">sc_spatial</span>(</span>
<span id="cb5-3"><a href="quantification-of-cell-states-using-svp.html#cb5-3" tabindex="-1"></a>     <span class="at">features =</span> <span class="fu">sort</span>(<span class="fu">names</span>(markerList)),</span>
<span id="cb5-4"><a href="quantification-of-cell-states-using-svp.html#cb5-4" tabindex="-1"></a>     <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x,  <span class="at">y =</span> y),</span>
<span id="cb5-5"><a href="quantification-of-cell-states-using-svp.html#cb5-5" tabindex="-1"></a>     <span class="at">geom =</span> geom_bgpoint,</span>
<span id="cb5-6"><a href="quantification-of-cell-states-using-svp.html#cb5-6" tabindex="-1"></a>     <span class="at">pointsize =</span> <span class="fl">1.2</span>,</span>
<span id="cb5-7"><a href="quantification-of-cell-states-using-svp.html#cb5-7" tabindex="-1"></a>     <span class="at">ncol =</span> <span class="dv">4</span></span>
<span id="cb5-8"><a href="quantification-of-cell-states-using-svp.html#cb5-8" tabindex="-1"></a>   ) <span class="sc">+</span></span>
<span id="cb5-9"><a href="quantification-of-cell-states-using-svp.html#cb5-9" tabindex="-1"></a>   <span class="fu">scale_color_viridis_c</span>()</span>
<span id="cb5-10"><a href="quantification-of-cell-states-using-svp.html#cb5-10" tabindex="-1"></a><span class="co">#&gt; Scale for colour is already present.</span></span>
<span id="cb5-11"><a href="quantification-of-cell-states-using-svp.html#cb5-11" tabindex="-1"></a><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="SVP_files/figure-html/heatmap_celltype-1.png" alt="(ref:heatmap_celltype)" width="1056" />
<p class="caption">
(#fig:heatmap_celltype)(ref:heatmap_celltype)
</p>
</div>
</div>
<div id="quantification-of-cell-type-using-the-reference-single-cell-data" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Quantification of cell-type using the reference single-cell data<a href="quantification-of-cell-states-using-svp.html#quantification-of-cell-type-using-the-reference-single-cell-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This approach requires obtaining the cell-type marker genes from the reference single-cell data. First, the reference single cell data should be normalized. Here, the command of <code>logNormCounts</code> of <code>scuttle</code> is used. Next, the <code>MCA</code> dimensionality reduction is performed by <code>runMCA</code> of <code>SVP</code>. Finally, the marker gene sets of cell-type are extracted via <code>runDetectMarker</code> of <code>SVP</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="quantification-of-cell-states-using-svp.html#cb6-1" tabindex="-1"></a><span class="co"># load the reference single cell transcriptome</span></span>
<span id="cb6-2"><a href="quantification-of-cell-states-using-svp.html#cb6-2" tabindex="-1"></a><span class="co"># it is a SingleCellExperiment</span></span>
<span id="cb6-3"><a href="quantification-of-cell-states-using-svp.html#cb6-3" tabindex="-1"></a>pdac_a_sce <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">url</span>(<span class="st">&quot;https://raw.githubusercontent.com/xiangpin/Data_for_Vignette_SVP/main/data/pdac_a_sce.rds&quot;</span>))</span>
<span id="cb6-4"><a href="quantification-of-cell-states-using-svp.html#cb6-4" tabindex="-1"></a>pdac_a_sce</span>
<span id="cb6-5"><a href="quantification-of-cell-states-using-svp.html#cb6-5" tabindex="-1"></a><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span id="cb6-6"><a href="quantification-of-cell-states-using-svp.html#cb6-6" tabindex="-1"></a><span class="co">#&gt; dim: 15418 1926 </span></span>
<span id="cb6-7"><a href="quantification-of-cell-states-using-svp.html#cb6-7" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb6-8"><a href="quantification-of-cell-states-using-svp.html#cb6-8" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb6-9"><a href="quantification-of-cell-states-using-svp.html#cb6-9" tabindex="-1"></a><span class="co">#&gt; rownames(15418): 1-Mar 1-Sep ... ZZEF1 ZZZ3</span></span>
<span id="cb6-10"><a href="quantification-of-cell-states-using-svp.html#cb6-10" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb6-11"><a href="quantification-of-cell-states-using-svp.html#cb6-11" tabindex="-1"></a><span class="co">#&gt; colnames(1926): Cell1 Cell2 ... Cell1925 Cell1926</span></span>
<span id="cb6-12"><a href="quantification-of-cell-states-using-svp.html#cb6-12" tabindex="-1"></a><span class="co">#&gt; colData names(1): Cell</span></span>
<span id="cb6-13"><a href="quantification-of-cell-states-using-svp.html#cb6-13" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(0):</span></span>
<span id="cb6-14"><a href="quantification-of-cell-states-using-svp.html#cb6-14" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb6-15"><a href="quantification-of-cell-states-using-svp.html#cb6-15" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span>
<span id="cb6-16"><a href="quantification-of-cell-states-using-svp.html#cb6-16" tabindex="-1"></a></span>
<span id="cb6-17"><a href="quantification-of-cell-states-using-svp.html#cb6-17" tabindex="-1"></a>pdac_a_sce <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(pdac_a_sce)</span>
<span id="cb6-18"><a href="quantification-of-cell-states-using-svp.html#cb6-18" tabindex="-1"></a></span>
<span id="cb6-19"><a href="quantification-of-cell-states-using-svp.html#cb6-19" tabindex="-1"></a><span class="co"># obtain the top high variable genes</span></span>
<span id="cb6-20"><a href="quantification-of-cell-states-using-svp.html#cb6-20" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb6-21"><a href="quantification-of-cell-states-using-svp.html#cb6-21" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(pdac_a_sce)</span>
<span id="cb6-22"><a href="quantification-of-cell-states-using-svp.html#cb6-22" tabindex="-1"></a>hvgs <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(stats)</span>
<span id="cb6-23"><a href="quantification-of-cell-states-using-svp.html#cb6-23" tabindex="-1"></a></span>
<span id="cb6-24"><a href="quantification-of-cell-states-using-svp.html#cb6-24" tabindex="-1"></a><span class="co"># perform the MCA reduction.</span></span>
<span id="cb6-25"><a href="quantification-of-cell-states-using-svp.html#cb6-25" tabindex="-1"></a>pdac_a_sce <span class="ot">&lt;-</span> pdac_a_sce <span class="sc">|&gt;</span> <span class="fu">runMCA</span>(<span class="at">assay.type =</span> <span class="st">&#39;logcounts&#39;</span>, <span class="at">subset.row=</span>hvgs[<span class="fu">seq</span>(<span class="dv">2000</span>)])</span>
<span id="cb6-26"><a href="quantification-of-cell-states-using-svp.html#cb6-26" tabindex="-1"></a><span class="co">#&gt; Computing Fuzzy Matrix</span></span>
<span id="cb6-27"><a href="quantification-of-cell-states-using-svp.html#cb6-27" tabindex="-1"></a><span class="co">#&gt; Computing SVD</span></span>
<span id="cb6-28"><a href="quantification-of-cell-states-using-svp.html#cb6-28" tabindex="-1"></a><span class="co">#&gt; Computing Coordinates</span></span>
<span id="cb6-29"><a href="quantification-of-cell-states-using-svp.html#cb6-29" tabindex="-1"></a></span>
<span id="cb6-30"><a href="quantification-of-cell-states-using-svp.html#cb6-30" tabindex="-1"></a><span class="co"># obtain the marker gene sets from reference single-cell transcriptome based on MCA space</span></span>
<span id="cb6-31"><a href="quantification-of-cell-states-using-svp.html#cb6-31" tabindex="-1"></a>refmakergene <span class="ot">&lt;-</span> <span class="fu">runDetectMarker</span>(pdac_a_sce, <span class="at">group.by=</span><span class="st">&#39;Cell&#39;</span>, <span class="at">ntop=</span><span class="dv">200</span>, <span class="at">present.prop.in.group=</span>.<span class="dv">2</span>, <span class="at">present.prop.in.sample=</span>.<span class="dv">5</span>)</span>
<span id="cb6-32"><a href="quantification-of-cell-states-using-svp.html#cb6-32" tabindex="-1"></a>refmakergene <span class="sc">|&gt;</span> <span class="fu">lapply</span>(length) <span class="sc">|&gt;</span> <span class="fu">unlist</span>()</span>
<span id="cb6-33"><a href="quantification-of-cell-states-using-svp.html#cb6-33" tabindex="-1"></a><span class="co">#&gt;                           Acinar cells          Ductal - terminal ductal like </span></span>
<span id="cb6-34"><a href="quantification-of-cell-states-using-svp.html#cb6-34" tabindex="-1"></a><span class="co">#&gt;                                     53                                     82 </span></span>
<span id="cb6-35"><a href="quantification-of-cell-states-using-svp.html#cb6-35" tabindex="-1"></a><span class="co">#&gt; Ductal - CRISP3 high/centroacinar like                         Cancer clone A </span></span>
<span id="cb6-36"><a href="quantification-of-cell-states-using-svp.html#cb6-36" tabindex="-1"></a><span class="co">#&gt;                                     90                                     90 </span></span>
<span id="cb6-37"><a href="quantification-of-cell-states-using-svp.html#cb6-37" tabindex="-1"></a><span class="co">#&gt;                  Ductal - MHC Class II                         Cancer clone B </span></span>
<span id="cb6-38"><a href="quantification-of-cell-states-using-svp.html#cb6-38" tabindex="-1"></a><span class="co">#&gt;                                     72                                     90 </span></span>
<span id="cb6-39"><a href="quantification-of-cell-states-using-svp.html#cb6-39" tabindex="-1"></a><span class="co">#&gt;                                 mDCs A            Ductal - APOL1 high/hypoxic </span></span>
<span id="cb6-40"><a href="quantification-of-cell-states-using-svp.html#cb6-40" tabindex="-1"></a><span class="co">#&gt;                                     89                                     92 </span></span>
<span id="cb6-41"><a href="quantification-of-cell-states-using-svp.html#cb6-41" tabindex="-1"></a><span class="co">#&gt;                             Tuft cells                                 mDCs B </span></span>
<span id="cb6-42"><a href="quantification-of-cell-states-using-svp.html#cb6-42" tabindex="-1"></a><span class="co">#&gt;                                     87                                     71 </span></span>
<span id="cb6-43"><a href="quantification-of-cell-states-using-svp.html#cb6-43" tabindex="-1"></a><span class="co">#&gt;                                   pDCs                        Endocrine cells </span></span>
<span id="cb6-44"><a href="quantification-of-cell-states-using-svp.html#cb6-44" tabindex="-1"></a><span class="co">#&gt;                                     50                                     60 </span></span>
<span id="cb6-45"><a href="quantification-of-cell-states-using-svp.html#cb6-45" tabindex="-1"></a><span class="co">#&gt;                      Endothelial cells                          Macrophages A </span></span>
<span id="cb6-46"><a href="quantification-of-cell-states-using-svp.html#cb6-46" tabindex="-1"></a><span class="co">#&gt;                                     81                                     76 </span></span>
<span id="cb6-47"><a href="quantification-of-cell-states-using-svp.html#cb6-47" tabindex="-1"></a><span class="co">#&gt;                             Mast cells                          Macrophages B </span></span>
<span id="cb6-48"><a href="quantification-of-cell-states-using-svp.html#cb6-48" tabindex="-1"></a><span class="co">#&gt;                                    120                                     48 </span></span>
<span id="cb6-49"><a href="quantification-of-cell-states-using-svp.html#cb6-49" tabindex="-1"></a><span class="co">#&gt;                     T cells &amp; NK cells                              Monocytes </span></span>
<span id="cb6-50"><a href="quantification-of-cell-states-using-svp.html#cb6-50" tabindex="-1"></a><span class="co">#&gt;                                     39                                     56 </span></span>
<span id="cb6-51"><a href="quantification-of-cell-states-using-svp.html#cb6-51" tabindex="-1"></a><span class="co">#&gt;                                   RBCs                            Fibroblasts </span></span>
<span id="cb6-52"><a href="quantification-of-cell-states-using-svp.html#cb6-52" tabindex="-1"></a><span class="co">#&gt;                                      9                                     97</span></span></code></pre></div>
<p>Then, the quantification of the cell-type gene signatures can also be performed via <code>runSGSA</code> with the marker gene sets.</p>
<p>(ref:runSGSA_ref) <strong>The pie plot of cell-type activity with the marker gene sets from reference single-cell transcriptome</strong></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="quantification-of-cell-states-using-svp.html#cb7-1" tabindex="-1"></a><span class="co"># use the maker gene from the reference single-cell transcriptome</span></span>
<span id="cb7-2"><a href="quantification-of-cell-states-using-svp.html#cb7-2" tabindex="-1"></a>pdac_a_spe <span class="ot">&lt;-</span> <span class="fu">runSGSA</span>(</span>
<span id="cb7-3"><a href="quantification-of-cell-states-using-svp.html#cb7-3" tabindex="-1"></a>  pdac_a_spe,</span>
<span id="cb7-4"><a href="quantification-of-cell-states-using-svp.html#cb7-4" tabindex="-1"></a>  <span class="at">gset.idx.list =</span> refmakergene,</span>
<span id="cb7-5"><a href="quantification-of-cell-states-using-svp.html#cb7-5" tabindex="-1"></a>  <span class="at">gsvaExp.name =</span> <span class="st">&#39;CellTypeRef&#39;</span></span>
<span id="cb7-6"><a href="quantification-of-cell-states-using-svp.html#cb7-6" tabindex="-1"></a>)</span>
<span id="cb7-7"><a href="quantification-of-cell-states-using-svp.html#cb7-7" tabindex="-1"></a><span class="co">#&gt; Building the nearest neighbor graph with the distance between features and</span></span>
<span id="cb7-8"><a href="quantification-of-cell-states-using-svp.html#cb7-8" tabindex="-1"></a><span class="co">#&gt; cells ...</span></span>
<span id="cb7-9"><a href="quantification-of-cell-states-using-svp.html#cb7-9" tabindex="-1"></a><span class="co">#&gt; elapsed time is 6.065000 seconds</span></span>
<span id="cb7-10"><a href="quantification-of-cell-states-using-svp.html#cb7-10" tabindex="-1"></a><span class="co">#&gt; Building the seed matrix using the gene set and the nearest neighbor graph for</span></span>
<span id="cb7-11"><a href="quantification-of-cell-states-using-svp.html#cb7-11" tabindex="-1"></a><span class="co">#&gt; random walk with restart ...</span></span>
<span id="cb7-12"><a href="quantification-of-cell-states-using-svp.html#cb7-12" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.020000 seconds</span></span>
<span id="cb7-13"><a href="quantification-of-cell-states-using-svp.html#cb7-13" tabindex="-1"></a><span class="co">#&gt; Calculating the affinity score using random walk with restart ...</span></span>
<span id="cb7-14"><a href="quantification-of-cell-states-using-svp.html#cb7-14" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.228000 seconds</span></span>
<span id="cb7-15"><a href="quantification-of-cell-states-using-svp.html#cb7-15" tabindex="-1"></a><span class="co">#&gt; Tidying the result of random walk with restart ...</span></span>
<span id="cb7-16"><a href="quantification-of-cell-states-using-svp.html#cb7-16" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.010000 seconds</span></span>
<span id="cb7-17"><a href="quantification-of-cell-states-using-svp.html#cb7-17" tabindex="-1"></a><span class="co">#&gt; ORA analysis ...</span></span>
<span id="cb7-18"><a href="quantification-of-cell-states-using-svp.html#cb7-18" tabindex="-1"></a><span class="co">#&gt; elapsed time is 0.012000 seconds</span></span>
<span id="cb7-19"><a href="quantification-of-cell-states-using-svp.html#cb7-19" tabindex="-1"></a></span>
<span id="cb7-20"><a href="quantification-of-cell-states-using-svp.html#cb7-20" tabindex="-1"></a>pdac_a_spe <span class="sc">|&gt;</span> </span>
<span id="cb7-21"><a href="quantification-of-cell-states-using-svp.html#cb7-21" tabindex="-1"></a>  <span class="fu">gsvaExp</span>(<span class="st">&#39;CellTypeRef&#39;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb7-22"><a href="quantification-of-cell-states-using-svp.html#cb7-22" tabindex="-1"></a>  <span class="fu">sc_spatial</span>(</span>
<span id="cb7-23"><a href="quantification-of-cell-states-using-svp.html#cb7-23" tabindex="-1"></a>    <span class="at">features =</span> <span class="fu">sort</span>(<span class="fu">names</span>(refmakergene)), </span>
<span id="cb7-24"><a href="quantification-of-cell-states-using-svp.html#cb7-24" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y),</span>
<span id="cb7-25"><a href="quantification-of-cell-states-using-svp.html#cb7-25" tabindex="-1"></a>    <span class="at">plot.pie =</span> <span class="cn">TRUE</span>, </span>
<span id="cb7-26"><a href="quantification-of-cell-states-using-svp.html#cb7-26" tabindex="-1"></a>    <span class="at">color =</span> <span class="cn">NA</span>, </span>
<span id="cb7-27"><a href="quantification-of-cell-states-using-svp.html#cb7-27" tabindex="-1"></a>    <span class="at">pie.radius.scale =</span> .<span class="dv">9</span></span>
<span id="cb7-28"><a href="quantification-of-cell-states-using-svp.html#cb7-28" tabindex="-1"></a>  )</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="SVP_files/figure-html/runSGSA_ref-1.png" alt="(ref:runSGSA_ref)" width="864" />
<p class="caption">
(#fig:runSGSA_ref)(ref:runSGSA_ref)
</p>
</div>
<p>We can found that the result based on marker gene sets from reference single-cell transcriptome is similar with the result based on marker gene sets from previous reports.</p>
</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-PDAC" class="csl-entry">
Moncada, Reuben, Dalia Barkley, Florian Wagner, Marta Chiodin, Joseph C Devlin, Maayan Baron, Cristina H Hajdu, Diane M Simeone, and Itai Yanai. 2020. <span>Integrating Microarray-Based Spatial Transcriptomics and Single-Cell RNA-Seq Reveals Tissue Architecture in Pancreatic Ductal Adenocarcinomas.</span> <em>Nature Biotechnology</em> 38 (3): 33342.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="overview-of-svp.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatial-statistical-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
